lib/core/utils/flattened-tree.js

```diff
@@ -46,6 +46,18 @@ function virtualDOMfromNode(node, shadowId) {
 				);
 			}
 			return vNodeCache._tabbableElements;
+		},
+
+		// abstract Node and Element APIs so we can run axe in DOM-less
+		// environments. these are static properties in the assumption
+		// that axe does not change any of them while it runs.
+		elementNodeType: node.nodeType,
+		elementNodeName: node.nodeName,
+		elementClassName: node.className,
```

Comment:
What about using a `hasClass()` type method instead of directly accessing the `className` property?